File: source/go/cli_tool.go
package main

import "fmt"

type Config struct {
	verbose bool
	output  string
	format  string
	count   int
}

func parseOption(key, value string, cfg *Config) {
	switch key {
	case "--verbose", "-v":
		cfg.verbose = true
	case "--output", "-o":
		cfg.output = value
	case "--format", "-f":
		cfg.format = value
	case "--count", "-n":
		n := 0
		fmt.Sscanf(value, "%d", &n)
		cfg.count = n
	}
}

func main() {
	toolName := "hemlock-tool"
	version := "1.0.0"

	cfg := &Config{
		verbose: false,
		output:  "stdout",
		format:  "text",
		count:   10,
	}

	simulatedArgs := [][2]string{
		{"--verbose", ""},
		{"--output", "result.txt"},
		{"--format", "json"},
		{"--count", "25"},
	}

	for _, arg := range simulatedArgs {
		parseOption(arg[0], arg[1], cfg)
	}

	fmt.Printf("%s v%s\n", toolName, version)
	fmt.Println("Parsed configuration:")
	fmt.Printf("  verbose: %v\n", cfg.verbose)
	fmt.Printf("  output:  %s\n", cfg.output)
	fmt.Printf("  format:  %s\n", cfg.format)
	fmt.Printf("  count:   %d\n", cfg.count)
	fmt.Println("Ready to process.")
}

File: source/rust/cli_tool.rs
struct Config {
    verbose: bool,
    output: String,
    format: String,
    count: i32,
}

impl Config {
    fn parse_option(&mut self, key: &str, value: &str) {
        match key {
            "--verbose" | "-v" => self.verbose = true,
            "--output" | "-o" => self.output = value.to_string(),
            "--format" | "-f" => self.format = value.to_string(),
            "--count" | "-n" => self.count = value.parse().unwrap_or(self.count),
            _ => {}
        }
    }
}

fn main() {
    let tool_name = "hemlock-tool";
    let version = "1.0.0";

    let mut config = Config {
        verbose: false,
        output: "stdout".to_string(),
        format: "text".to_string(),
        count: 10,
    };

    let simulated_args = [
        ("--verbose", ""),
        ("--output", "result.txt"),
        ("--format", "json"),
        ("--count", "25"),
    ];

    for (key, val) in &simulated_args {
        config.parse_option(key, val);
    }

    println!("{} v{}", tool_name, version);
    println!("Parsed configuration:");
    println!("  verbose: {}", config.verbose);
    println!("  output:  {}", config.output);
    println!("  format:  {}", config.format);
    println!("  count:   {}", config.count);
    println!("Ready to process.");
}
